src/coreclr/jit/fgbasic.cpp

```diff
@@ -4893,6 +4900,11 @@ BasicBlock* Compiler::fgSplitBlockBeforeTree(
     block->bbFlags |=
         originalFlags & (BBF_SPLIT_GAINED | BBF_IMPORTED | BBF_GC_SAFE_POINT | BBF_LOOP_PREHEADER | BBF_RETLESS_CALL);
 
+    // The above flag propagation will unset BBF_NONE_QUIRK, which should be set for prevBb
+    // since it jumps to its new next block.
+    assert(prevBb->KindIs(BBJ_ALWAYS) && prevBb->JumpsToNext());
+    prevBb->bbFlags |= BBF_NONE_QUIRK;
```

Comment:
`BBF_NONE_QUIRK` should likely be part of `BBF_SPLIT_GAINED`.

Reply:
(Or otherwise you may want to double check uses of `BBF_SPLIT_GAINED` -- presumably they should all have this treatment.)

Reply:
I did a quick look, and I think your suggestion works for all of the (few) uses of `BBF_SPLIT_GAINED`. I'll verify locally first

Reply:
Actually, after looking again, is the comment here right? The above masking should not be removing `BBF_NONE_QUIRK` from `prevBb` from what I can tell (but it definitely is right that it needs to unconditionally add the flag to it).

Reply:
Looks like I was thinking shortsightedly with that comment; if `BBF_NONE_QUIRK` was not set before splitting the block, then it will need to be set for `prevBb` (and it's cheaper to unconditionally set `BBF_NONE_QUIRK` than to do any additional check). I'll remove this comment.